File: Julia_Experiences/Trainings/Bindings/Python/math_module.py
# -*- coding: utf-8 -*-
# math_module.py

import numpy as np


def sum_squares(n:int) -> float:
    result = 0.0
    for i in range(1, n + 1):
        result += float(i)**2
    return result

def sum_squares_np(n:int) -> float:
    return np.sum(np.arange(1, n + 1, dtype=np.float64)**2)

if __name__ == '__main__':
    assert sum_squares(2)  == 5
    assert sum_squares(10) == 385
    assert sum_squares_np(10) == 385

File: Julia_Experiences/Trainings/Bindings/Python/math_module.jl
# math_module.jl

module MathModule

    export sum_squares

    function sum_squares(n::Int)::Float64
        result = 0.0
        # Use Single Instruction, Multiple Data
        @simd for i in 1:n
            @inbounds result += Float64(i)^2
        end
        return result
    end

end

if abspath(PROGRAM_FILE) == @__FILE__

    @assert MathModule.sum_squares(2)  == 5
    @assert MathModule.sum_squares(10) == 385

end
